mypyc/irbuild/for_helpers.py

```diff
@@ -170,6 +177,16 @@ def make_for_loop_generator(builder: IRBuilder,
         for_list.init(expr_reg, target_type, reverse=False)
         return for_list
 
+    # Unfortunately fast dict iteration breaks too many tests on Python 3.5.
```

Comment:
This is not a big problem, since Python 3.5 will reach EoL relatively soon.

Reply:
There is actually a bigger issue Sully found: even on newer version this is not safe for some dict subclasses. For example if someone passes an `OrderedDict`, after `.move_to_end()` was called on it, it may not always work properly w.r.t. items order.

I will need to investigate this more on weekend (this applies to other dict PR too).